### PR TITLE
Add TagHelper Code generation support

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Issue - #882 

@NTaylorMullen @rynowak

### CODE REVIEW

File: src/Microsoft.AspNetCore.Razor.Evolution/DefaultRazorRuntimeCSharpLoweringPhase.cs [hunk 1]
```diff
@@ -34,6 +34,13 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument)
                 LineMappings = renderingContext.Writer.LineMappingManager.Mappings,
             };
 
+            if (renderingContext.ErrorSink.Errors.Count > 0)
+            {
+                var combinedErrors = syntaxTree.Diagnostics.Concat(renderingContext.ErrorSink.Errors).ToList();
+                syntaxTree = RazorSyntaxTree.Create(syntaxTree.Root, combinedErrors, syntaxTree.Options);
+            }
+
+            codeDocument.SetSyntaxTree(syntaxTree);
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Razor.Evolution/DefaultRazorRuntimeCSharpLoweringPhase.cs]: We shouldn't be adding errors to the syntax tree in a later phase. There errors belong on the CSharp document

File: test/Microsoft.AspNetCore.Razor.Evolution.Test/TestFiles/IntegrationTests/RuntimeCodeGenerationIntegrationTest/NestedTagHelpers.codegen.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+                    __backed__tagHelperScopeManager = new Microsoft.AspNetCore.Razor.Runtime.TagHelperScopeManager(StartTagHelperWritingScope, EndTagHelperWritingScope);
+                }
+                return __backed__tagHelperScopeManager;
+            }
+        }
+        private global::PTagHelper __PTagHelper = null;
+        private global::FormTagHelper __FormTagHelper = null;
+        private global::InputTagHelper __InputTagHelper = null;
+        #pragma warning disable 1998
+           ()
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Razor.Evolution.Test/TestFiles/IntegrationTests/RuntimeCodeGenerationIntegrationTest/NestedTagHelpers.codegen.cs]: Will need to add the test adapter to fill out the class/namespace etc. names

**Comment by ajaybhargavb** [comment]
🆙 📅 No longer in design

File: test/Microsoft.AspNetCore.Razor.Evolution.Test/TestFiles/IntegrationTests/RuntimeCodeGenerationIntegrationTest/ComplexTagHelpers.cshtml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            else
+            {
+                <p>Current Time: <input type=@checkbox checked=true /></p>
+                <input tYPe='@(true ? "checkbox" : "anything")' />
+                <input type='@if(true) { <text>checkbox</text> } else { <text>anything</text> }'>
+            }
+        </p>
+        <p unbound="first value" age="@DateTimeOffset.Now.Year - 1970" unbound="second value" >
+            @{ var @object = false;}
+            <input ChecKED="@(@object)">
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNetCore.Razor.Evolution.Test/TestFiles/IntegrationTests/RuntimeCodeGenerationIntegrationTest/ComplexTagHelpers.cshtml]: This doesn't look to be rendering correctly compared to old world. Also talked offline about combing through a few of these complex ones just to ensure things like this aren't missed.

**Comment by ajaybhargavb** [comment]
🆙 📅

**Pull request merged**